spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java

```diff
@@ -76,4 +79,19 @@ public GcpProjectIdProvider gcpProjectIdProvider() {
 				? () -> this.gcpProperties.getProjectId()
 				: new DefaultGcpProjectIdProvider();
 	}
+
+	/**
+	 * Google Client Library APIs (i.e., non-Veneer/google-cloud-java) require
+	 * {@link GoogleCredential} instead of {@link GoogleCredentials}.
+	 */
+	@Bean
+	@ConditionalOnMissingBean
+	public Supplier<GoogleCredential> googleCredentialSupplier() throws IOException {
```

Comment:
??? It's not good. No body would be happy to deal with the `Supplier` bean. With this you try to eliminate the good search improvement via `BeanFactory.getBean(Class<T> requiredType)`